peer.go

```diff
@@ -331,6 +331,9 @@ func (p *Peer) Exec(ctx context.Context, method string, data []byte) ([]byte, er
 	if !ok {
 		return nil, errUnknownMethod{}
 	}
+	if ContextPeer(ctx) == nil {
```

Comment:
Hmm, I'm not sure what the semantics should be here: to the handler being invoked by Exec, the peer is _always_ local, no?

I can see how passing along a remote peer would help for doing "invoke helper function" type stuff, but for other kinds of helpers (e.g. stream from a local handler and then filter the output before streaming it onwards) the lack of takeover is equally odd.

I dunno, not sure what makes the most sense. I defer to you, although either way I think the docstring for Exec should specify, because I don't think either is obviously always right.

Reply:
The peer is always local regardless, there's no way in general to get a handle on the remote peer (though in theory one could do it if both are in the same process). But the condition here is so that if you are already in a handler and the peer is set, we don't introduce another layer of context

Reply:
oh, derp, right, I was confused. I thought somehow that there was a Peer that could be set to represent "localhost", where further calls would loop back as-if made remotely. But I see now that makes no sense.

Reply:
There isn't, but that is the effect Exec is meant to allow.